### PR TITLE
Track colony versions in the database

### LINKED ISSUE #102: Track Colony version in the database
This issue covers the work needed to store and retrieve the Colony version in the database.
It includes but is not limited to:
 - Modifying the schema/queries/mutations
 - Handling `ColonyUpgraded` events in `block-ingestor`
 - Update existing isSupportedColonyVersion checks to use the Colony version from the database

### PR DESCRIPTION
**Pull request by jakubcolony**
## Description

This PR adds support for tracking colony version in the block-ingestor. See [block-ingestor#4](https://github.com/JoinColony/block-ingestor/pull/4) for details and testing.

Summary of CDapp changes:
- New `version` field on Colony model
- Renamed `item` field of CurrentVersion model to `key` to avoid confusion with `items`
- Adding `version` field to all existing create colony mutations
- Adding `isSupportedColonyVersion` checks to extensions

Resolves #102

### CODE REVIEW

File: src/components/common/Extensions/ExtensionActionButton/ExtensionActionButton.tsx [hunk 1]
```diff
@@ -39,6 +39,8 @@ const ExtensionActionButton = ({ extensionData }: Props) => {
     );
   };
 
+  const isSupportedColonyVersion = colony.version >= 5;
```
Comment (jakubcolony) [inline src/components/common/Extensions/ExtensionActionButton/ExtensionActionButton.tsx:42]: There used to be a `ColonyVersion` enum exported from `colony-js` but it's not there anymore. Open to suggestions on replacing the hardcoded value

Comment (willm30) [reply src/components/common/Extensions/ExtensionActionButton/ExtensionActionButton.tsx:42]: I'd say have a constant 
``` js
const MIN_SUPPORTED_COLONY_VERSION = 5
const isSupportedColonyVersion = colony.version >= MIN_SUPPORTED_COLONY_VERSION;
```
or similar

File: src/components/common/Extensions/ExtensionUpgradeButton/ExtensionUpgradeButton.tsx [hunk 2]
```diff
@@ -28,19 +29,18 @@ const ExtensionUpgradeButton = ({ extensionData }: Props) => {
...
+    return null;
+  }
+
+  const isSupportedColonyVersion = colony.version >= 5;
 
   const extensionCompatible = isExtensionCompatible(
     Extension[extensionData.extensionId],
-    extensionData.availableVersion as ExtensionVersion,
-    10,
+    (extensionData.availableVersion + 1) as ExtensionVersion,
```
Comment (willm30) [inline src/components/common/Extensions/ExtensionUpgradeButton/ExtensionUpgradeButton.tsx:40]: How come we're checking if availableVersion + 1 is compatible? Isn't availableVersion the latest version in the db (i.e. latest + 1 doesn't exist)?

File: scripts/temp-create-data.js [hunk 3]
```diff
@@ -230,6 +230,7 @@ const createMetacolony = async (singerOrWallet) => {
         name: 'meta',
         profile: { displayName: 'Metacolony' },
         type: 'METACOLONY',
+        version: BigNumber.from(metacolonyVersion).toNumber()
```
Comment (willm30) [inline scripts/temp-create-data.js:233]: `BigNumber.from...toNumber()` could live as a little utility e.g.

numbers.ts
``` js
const toNumber = (bigNumber) => BigNumber.from(bigNumber).toNumber()
```

**Review by willm30 — APPROVED** [review]
Looks good 👍

File: src/components/common/Extensions/ExtensionUpgradeButton/ExtensionUpgradeButton.tsx (hunk 2 above)
Comment (jakubcolony) [reply src/components/common/Extensions/ExtensionUpgradeButton/ExtensionUpgradeButton.tsx:40]: Good spot, my bad!

File: scripts/temp-create-data.js (hunk 3 above)
Comment (jakubcolony) [reply scripts/temp-create-data.js:233]: Makes sense 👍

File: src/redux/sagas/colony/colonyCreate.ts [hunk 4]
```diff
@@ -304,6 +305,7 @@ function* colonyCreate({
             name: givenColonyName,
             colonyNativeTokenId: tokenAddress,
             profile: { displayName },
+            version: BigNumber.from(currentColonyVersion).toNumber(),
```
[the remark this answers is not in the document]
Comment (jakubcolony) [reply src/redux/sagas/colony/colonyCreate.ts:308]: I'm thinking that this would be covered by #179

**Review by rdig — APPROVED** [review]
Yep, this looks very clean. Can't really find anything I'd want to change, but my review here is late, after you've addressed Will's comments.

Anyway, nice one :100:

**Pull request merged**